src/renderer/components/command-palette/command-container.tsx

```diff
@@ -0,0 +1,96 @@
+
+import "./command-container.scss";
+import { action, observable } from "mobx";
+import { observer } from "mobx-react";
+import React from "react";
+import { Dialog } from "../dialog";
+import { EventEmitter } from "../../../common/event-emitter";
+import { subscribeToBroadcast } from "../../../common/ipc";
+import { CommandDialog } from "./command-dialog";
+import { CommandRegistration, commandRegistry } from "../../../extensions/registries/command-registry";
+import { clusterStore } from "../../../common/cluster-store";
+import { workspaceStore } from "../../../common/workspace-store";
+import { Cluster } from "../../../main/cluster";
+
+export type CommandDialogEvent = {
+  component: React.ReactElement
+};
+
+const commandDialogBus = new EventEmitter<[CommandDialogEvent]>();
+
+export class CommandOverlay {
+  static open(component: React.ReactElement) {
+    commandDialogBus.emit({ component });
+  }
+
+  static close() {
+    commandDialogBus.emit({ component: null });
+  }
+}
+
+@observer
+export class CommandContainer extends React.Component<{cluster?: Cluster}> {
+  @observable commandComponent: React.ReactElement;
+
+  private escHandler(event: KeyboardEvent) {
+    if (event.key === "Escape") {
+      event.stopPropagation();
+      this.closeDialog();
+    }
+  }
+
+  @action
+  private closeDialog() {
+    this.commandComponent = null;
+  }
+
+  private findCommandById(commandId: string) {
+    return commandRegistry.getItems().find((command) => command.id === commandId);
+  }
+
+  private runCommand(command: CommandRegistration) {
+    command.action({
+      cluster: clusterStore.active,
+      workspace: workspaceStore.currentWorkspace
+    });
+  }
```

Comment:
```suggestion
  static findCommandById(commandId: string) {
    return commandRegistry.getItems().find((command) => command.id === commandId);
  }

  static runCommand(commandId: string) {
    this.findCommandById(commandId)?.action({
      cluster: clusterStore.active,
      workspace: workspaceStore.currentWorkspace
    });
  }
```

Reply:
I prefer to keep these private (at least for now).

Reply:
You can make them `private static`

Reply:
@Nokel81 you cannot use `private static findCommandById` in `private static runCommand`

Reply:
Yes you can

https://www.typescriptlang.org/play?#code/MYGwhgzhAEBiD29oG8BQ0PQA4CcCWAbmAC4Cm0ExJew0AQmDgBQCUAXNAfHgCYqrpMAXwGZs+ImQpViNenggALVhy69+YsQngA6BsxYBuQRhEnoAO0Y5EAI3grO3Pmk2ZtehcqPmRQoA

Reply:
@Nokel81 yes, seems like it introduced in _Typescript 4.1+_, currently is:

![image](https://user-images.githubusercontent.com/6377066/106589634-e47d3080-6554-11eb-8088-5e7c976ec8da.png)

Reply:
Oh, right, I just checked in editor. Wrong assumption :)